src/geocel/vg/detail/SurfNavigator.hh

```diff
@@ -79,9 +88,9 @@ class SurfNavigator
             return step_limit;
         }
 
-        int exit_surf = 0;
-        auto step = vgbrep::protonav::ComputeStepAndHit(
-            globalpoint, globaldir, in_state, out_state, exit_surf, step_limit);
+	long hitsurf_id = 0;
+        auto step = vgbrep::protonav::BVHSurfNavigator<Precision>::ComputeStepAndNextSurface(
+            globalpoint, globaldir, in_state, out_state, hitsurf_id, step_limit);
```

Comment:
Something here is misbehaving compared to the volume version:
```
54: /Users/seth/Code/celeritas-temp/test/geocel/vg/Vecgeom.test.cc:164: Failure
54: Value of: next.boundary
54:   Actual: true
54: Expected: false
```
for the calls
```
    auto next = geo.find_next_step(from_cm(100));
    EXPECT_SOFT_EQ(100, to_cm(next.distance));
    EXPECT_FALSE(next.boundary);
```

So it's claiming to have a boundary for the next surface, even though it doesn't hit it within the step limit.